contracts/GenerativeArtworks.sol

```diff
@@ -110,12 +110,24 @@ contract GenerativeArtworks is ERC721Enumerable {
         pieces[pieceId].paused = !pieces[pieceId].paused;
     }
 
-    function addPiece(string memory pieceName) external onlyAdmin {
+    function addPiece(string memory name, string memory description, string memory license, string memory baseURI, uint256 maxPrints, string memory script, uint256 pricePerPrintInWei) external onlyAdmin returns (uint256) {
         uint256 pieceId = nextPieceId;
-        pieces[pieceId].name = pieceName;
-        pieces[pieceId].paused = true;
+        pieces[pieceId] = Piece({
+            name: name,
+            description: description,
+            license: license,
+            baseURI: baseURI,
+            currentPrints: 0,
+            maxPrints: maxPrints,
+            script: script,
+            active: false,
+            locked: false,
```

Comment:
active and locked are false by default so you don't technically need this. Should we remove it? Or does this need to be here because we're casting it?